### PR TITLE
modify the order of generating files

### LINKED ISSUE #2419: makefile: error when creating a new CRD
## Bug Report

**What version of Kubernetes are you using?**
Client Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.10-dispatcher", GitCommit:"f5757a1dee5a89cc5e29cd7159076648bf21a02b", GitTreeState:"clean", BuildDate:"2020-02-06T03:31:35Z", GoVersion:"go1.12.12b4", Compiler:"gc", Platform:"darwin/amd64"}

**What version of Chaos Mesh are you using?**
latest

**What did you do?**
create a new type CRD, execute "make generate".

**What did you expect to see?**
create successfully

**What did you see instead?**

it seems that something wrong with "generate-ctrl"

image
image

**Output of chaosctl**

### LINKED ISSUE #2594: `make generate should` not depend on `make generate-ctrl`
## Bug Report

**What version of Chaos Mesh are you using?**
latest
**What did you do?**
I tried to use `make generate` to generate `zz_generated.deepcopy.go` after changing a struct in `stresschaos_type.go`. 

**What did you expect to see?**
New `DeepCopyInto` method be generated.

**What did you see instead?**
image
`make generate` depends on `make generate-ctrl`, but `maek generate-ctrl` can not work when `zz_generated.deepcopy.go` is not generated.

### LINKED ISSUE #2644: does new experiment will generate new Deepcopy file when execute make generate？
## Question

Before asking a question, make sure you have:

- Reviewed relevant Kubernetes information: Google your error messages and look at K8s docs.
- Searched open and closed [GitHub issues](https://github.com/chaos-mesh/chaos-mesh/issues?utf8=%E2%9C%93&q=is%3Aissue)
- Read the documentation:
  - [Chaos Mesh Readme](https://github.com/chaos-mesh/chaos-mesh)
  - [Chaos Mesh Doc](https://chaos-mesh.org/docs)

’make generate‘ didn't update  deepcopy file

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### What problem does this PR solve?

close #2594 
close #2419 
close #2644

Fix the order of generating tasks.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -422,13 +422,16 @@ $(eval $(call RUN_IN_DEV_ENV_TEMPLATE,tidy,clean))
...
 
-define generate-make
+define generate-deepcopy-make
 	cd ./api/v1alpha1 ;\
 		controller-gen object:headerFile=../../hack/boilerplate/boilerplate.generatego.txt paths="./..." ;
 endef
-$(eval $(call RUN_IN_DEV_ENV_TEMPLATE,generate,chaos-build generate-ctrl swagger_spec))
+$(eval $(call RUN_IN_DEV_ENV_TEMPLATE,generate-deepcopy,chaos-build))
+
+generate: generate-ctrl swagger_spec generate-deepcopy chaos-build
```
Comment (STRRL) [inline Makefile:433]: Should we need to add `generate-deepcopy` as the dependency of `generate-ctrl`?

Does `generate-ctrl` depends on the `DeepCopy()` methods ? @Hexilee

Comment (YangKeao) [reply Makefile:433]: Actually, I never expect the `Makefile` under Chaos Mesh can run with `-j xx`, so the dependency is always built in order 😢 .

**Comment by YangKeao** [comment]
@STRRL @iguoyr @Hexilee PTAL

File: Makefile (hunk 1 above)
Comment (STRRL) [reply Makefile:433]: Oops actually I also never think about `-j`, I just notice the order in `generate-ctrl swagger_spec generate-deepcopy chaos-build`.

It seems `schema.resolvers.go` does not use any methods in `zz_generated.deepcopy.go`, `zz_generated.chaosmesh.go` and `zz_generated.<other>.chaosmesh.go`, I think `generate-ctrl` does not depend on `generate-deepcopy` and `chaos-builder`

**Review by STRRL — APPROVED** [review]
LGTM

**Review by iguoyr — APPROVED** [review]
LGTM

**Pull request merged**